waiter/src/waiter/auth/authentication.clj

```diff
@@ -52,23 +52,27 @@
                     {:status http-400-bad-request}))))
 
 (defn- add-cached-auth
-  [response password principal age-in-seconds]
-  (cookie-support/add-encoded-cookie response password AUTH-COOKIE-NAME [principal (tc/to-long (t/now))]
-                                     (or age-in-seconds (-> 1 t/days t/in-seconds))))
+  [response password principal age-in-seconds auth-metadata]
+  (let [cookie-value [principal (tc/to-long (t/now)) auth-metadata]
```

Comment:
Should we avoid adding `auth-metadata` to `cookie-value` if `auth-metadata` is nil?

Reply:
Avoided adding the metadata entry into the cookie.